pkg/cmd/extension/command.go

```diff
@@ -102,15 +105,38 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 					return err
 				}
 				if err := checkValidExtension(cmd.Root(), m, repo.RepoName()); err != nil {
+					// TODO i feel like this should check for a gh-foo script
 					return err
 				}
 
+				client, err := f.HttpClient()
+				if err != nil {
+					return fmt.Errorf("could not make http client: %w", err)
+				}
+				client = api.NewCachedClient(client, time.Second*30)
+
+				isBin, err := isBinExtension(client, repo)
+				if err != nil {
+					return fmt.Errorf("could not check for binary extension: %w", err)
+				}
+				if isBin {
+					return m.InstallBin(client, repo)
+				}
+
+				hs, err := hasScript(client, repo)
```

Comment:
What are your thoughts on moving this check above `isBinExtension`? Seems like we could then remove the `hasScript` check from `isBinExtension`. 

Or instead what are your thoughts on just encapsulating the type determination logic in its own function. Something like
```golang
type extensionType string

const (
	binary extensionType = "bin"
	git                  = "git"
)

func getExtensionType(client *http.Client, repo ghrepo.Interface) (extensionType, error) {
	hs, err := hasScript(client, repo)
	if err != nil {
		return "", err
	}
	if hs {
		return git, nil
	}
	
	_, err = fetchLatestRelease(client, repo)
 	if err != nil {
		return "", err
 	}
	return binary, nil
}
```

Reply:
I think my refactoring made this comment obsolete so I'm going to resolve but feel free to correct me~